src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -373,12 +330,12 @@ internal void AddToPendingTasks(Task task)
                 case TaskStatus.Canceled:
                     break;
                 case TaskStatus.Faulted:
-                    // We want to throw immediately if the task failed synchronously instead of
+                    // We want to immediately handle exceptions if the task failed synchronously instead of
                     // waiting for it to throw later. This can happen if the task is produced by
                     // an 'async' state machine (the ones generated using async/await) where even
                     // the synchronous exceptions will get captured and converted into a faulted
                     // task.
-                    ExceptionDispatchInfo.Capture(task.Exception.InnerException).Throw();
+                    HandleException(task.Exception.GetBaseException());
```

Comment:
Can you explain to me the difference between these two? I thought you had to do ExceptionDispatchInfo.Capture to preserve the original callstack